include/aikido/planner/ompl/detail/OMPLConfigurationToConfigurationPlanner-impl.hpp

```diff
@@ -0,0 +1,192 @@
+  auto sspace = ompl_make_shared<GeometricStateSpace>(
+      mStateSpace,
+      std::move(interpolator),
+      std::move(dmetric),
+      std::move(sampler),
+      std::move(boundsConstraint),
+      std::move(boundsProjector),
+      maxDistanceBetweenValidityChecks);
+
+  // Space Information
+  auto si = ompl_make_shared<::ompl::base::SpaceInformation>(sspace);
+
+  // Setup the planner
+  mPlanner = ompl_make_shared<PlannerType>(si);
+}
+
+//==============================================================================
+template <class PlannerType>
+trajectory::TrajectoryPtr
+OMPLConfigurationToConfigurationPlanner<PlannerType>::plan(
+    const SolvableProblem& problem, Result* result)
+{
+  auto si = mPlanner->getSpaceInformation();
+
+  // Only geometric statespaces are supported.
+  assert(ompl_dynamic_pointer_cast<GeometricStateSpace>(si->getStateSpace()));
+  auto sspace
+      = ompl_static_pointer_cast<GeometricStateSpace>(si->getStateSpace());
+
+  // Set validity checker.
+  std::vector<constraint::ConstTestablePtr> constraints{
+      problem.getConstraint(), sspace->getBoundsConstraint()};
+  auto conjunctionConstraint
+      = std::make_shared<constraint::TestableIntersection>(
+          mStateSpace, std::move(constraints));
+  ::ompl::base::StateValidityCheckerPtr vchecker
+      = ompl_make_shared<StateValidityChecker>(si, conjunctionConstraint);
+  si->setStateValidityChecker(vchecker);
+
+  ::ompl::base::MotionValidatorPtr mvalidator
+      = ompl_make_shared<MotionValidator>(
+          si, sspace->getMaxDistanceBetweenValidityChecks());
+  si->setMotionValidator(mvalidator);
+
+  // Define the OMPL problem.
+  auto pdef = ompl_make_shared<::ompl::base::ProblemDefinition>(si);
+
+  // Provide the problem definition with start and goal configuations.
+  auto start = sspace->allocState(problem.getStartState());
+  auto goal = sspace->allocState(problem.getGoalState());
+  pdef->setStartAndGoalStates(start, goal);
+
+  // pdef clones and maintains start and goal internally.
+  sspace->freeState(start);
+  sspace->freeState(goal);
+
+  // Solve the planning problem.
+  mPlanner->setProblemDefinition(pdef);
+  mPlanner->setup();
+
+  // TODO (avk): Introduce other termination conditions for planners (as in
+  // OMPL).
+  auto solved = mPlanner->solve(::ompl::base::plannerNonTerminatingCondition());
+
+  if (solved)
+  {
+    auto returnTraj = std::make_shared<trajectory::Interpolated>(
+        mStateSpace, sspace->getInterpolator());
+
+    // Get the path
+    auto path = ompl_dynamic_pointer_cast<::ompl::geometric::PathGeometric>(
+        pdef->getSolutionPath());
+    if (!path)
+    {
+      throw std::invalid_argument(
+          "Path is not of type PathGeometric. Cannot convert to aikido "
+          "Trajectory.");
+    }
+
+    for (std::size_t idx = 0; idx < path->getStateCount(); ++idx)
+    {
+      const auto* st
+          = static_cast<aikido::planner::ompl::GeometricStateSpace::StateType*>(
```

Comment:
Done. For now, I don't much of a preference on this, since we are only working with geometric planners/statespaces. But I guess when we later extend our support, we might be dealing with other state types. 

Thanks!